staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go

```diff
@@ -335,6 +335,9 @@ func (o *ExposeServiceOptions) RunExpose(cmd *cobra.Command, args []string) erro
 		}
 
 		if o.DryRunStrategy == cmdutil.DryRunClient {
+			if meta, err := meta.Accessor(object); err == nil && o.EnforceNamespace {
+				meta.SetNamespace(o.Namespace)
+			}
```

Comment:
@soltysh @deads2k Is it preferable to take this route, and manipulate the generated object before printing.. or would it be better to, [in the generator](https://github.com/kubernetes/kubectl/blob/master/pkg/generate/versioned/service.go#L80), add the namespace to the generated object?

Reply:
That code seems to be used in the ```run``` command as well.

https://github.com/kubernetes/kubernetes/blob/8b24d1f2f92063faa4cff1cdea6a3d504597b555/staging/src/k8s.io/kubectl/pkg/cmd/run/run.go#L677-L679

Reply:
This is ok temporary solution, we're slowly getting rid of generators which will clear this situation.